stdlib/in_channel.ml

```diff
@@ -63,4 +78,42 @@ let really_input_string ic len =
   | s -> Some s
   | exception End_of_file -> None
 
+let input_all ic =
+  let rec loop buf ofs =
+    let len = Bytes.length buf in
+    if ofs < len then
+      let r = Stdlib.input ic buf ofs (len - ofs) in
+      if r = 0 then
+        Bytes.sub_string buf 0 ofs
+      else
+        loop buf (ofs + r)
+    else begin (* ofs = len *)
+      match Stdlib.input_char ic with
+      | exception End_of_file ->
+          Bytes.unsafe_to_string buf
+      | c ->
```

Comment:
Something feels a little bit odd about this pattern. It seems to me that if we read from a channel that doesn't support `in_channel_length`, such as a pipe, we are going to alternate between reading a large block and reading 1 byte, while we could skip the 1 byte reads.

Reply:
You are right. I rearranged the code to only try the 1-byte read once, for the "fast path".